functions.py

```diff
@@ -170,10 +170,27 @@ def get_jenkins_job_info(server, job_name):
 	return jenkins_api_info
 
 
-def get_jenkins_jobs(server, job_search_fields):
+def get_jenkins_jobs(server, job_search_fields, job_search_fields_regex=None):
 	''' takes in a Jenkins server object and job_search_fields string
 		returns list of jobs with given search field as part of their name
 	'''
+	relevant_jobs = []
+
+	if job_search_fields_regex:
+		try:
+			re.compile(job_search_fields_regex)
+			# fetch all jobs from server that match the given regex
+			all_jobs = server.get_job_info_regex(job_search_fields_regex)
+
+			# parse out all jobs that do not contain any search field and/or are not OSP10, OSP13, OSP15 or OSP16 jobs
+			supported_versions = ['10', '13', '15', '16', '16.1']
+			for job in all_jobs:
+				job_name = job['name']
+				if any(supported_version in job_name for supported_version in supported_versions):
+					relevant_jobs.append(job)
+					print(job_name)
+		except re.error:
+			pass
```

Comment:
Now that I think about it, @nathan-weinberg  do we wan't to show any error msg or something like "Sorry, Invalid Regex" to the user

Reply:
Yes an error message such as below would be preferred

```suggestion
except re.error as e:
    print("Error compiling regex: {}".format(e))
```

Reply:
Furthermore, if this try/except block is only meant to handle the compilation of the the regex, the rest of the code currently inside it should instead **follow** the block

@agarcia64 What is the expected behavior if the compilation fails? If it just continues as is currently implemented, won't the program just crash?

Reply:
As it is now, if the compilation fails it'll just pass and continue while ignoring the. regex. However, if the code inside that block is moved outside, then invalid regex will crash the program. 

I'll replace the pass with the error message!

Reply:
So if it ignores the regex, what happens next?

Reply:
IMO we should just replace the fields config var with the regex var

Reply:
If the regex is ignored, it'll just move onto the pre-existing search using only job_search_fields. 

@amodi5 raises a good point, although I think it would be easier to keep the existing job_search_fields, delete the regex field and just check job_search_fields for valid regex. I tried doing this initially but couldn't get consistent results, however now that I've been working on this problem for a while I think I can get it done.